articles/arm-x86/index.md

```diff
@@ -0,0 +1,85 @@
+
+ARM is commonly used in smartphones. It's smaller, more energy-efficient, and creates less heat.  This makes it perfect for mobile devices. Laptops don't seem to be an exception to this. ARM would Apple to make their MacBooks cooler. They've historically had problems with this in [2020](https://www.macworld.co.uk/news/mac/2020-macbook-air-problems-3788127/), [2018](https://www.zdnet.com/article/apple-patches-2018-macbook-pro-to-address-throttled-performance-and-overheating/), [2015](https://www.theverge.com/2019/6/20/18693136/apple-recall-2015-15-inch-macbook-pro-battery-overheat-fire-risk-safety), and [older](https://discussions.apple.com/thread/5815813). Extending the battery life also sounds very nice. [There's also a possibility of Apple's laptops getting smaller](https://www.forbes.com/sites/ewanspence/2020/07/20/apple-macos-bigsur-macbook-pro-arm-intel-advantages-danger/#1354fad339f7). Although, consumers most likely won't care about that as much. Laptops can already be [very light](https://www.lg.com/us/laptops/lg-14z90n-u.aas7u1-ultra-slim-laptop).
+
+## Drawbacks of ARM
+
+There is an issue of compatibility here though. Programs will need to be completely rewritten to run on Apple's newer machines. This isn't too bad, since most compilers can target ARM just fine. Assembly developers are probably having a panic attack right now. Apple is also hoping that [Rosetta 2](https://www.theverge.com/21304182/apple-arm-mac-rosetta-2-emulation-app-converter-explainer) can run all the x86 applications. Rosetta 1 was known for being much slower than running on original hardware. This is by necessity. Rosetta needs to translate the x86 instructions into ARM instructions in real-time. To be fair, Java converts from [bytecode to everything else](https://www.javatpoint.com/java-bytecode), and it seems to work [fine-ish](https://benchmarksgame-team.pages.debian.net/benchmarksgame/fastest/java.html). Still, it seems hard to imagine being able to play games from the 2010's if Microsoft also decides to switch to ARM.
+
+There's also an issue of speed. Since ARM has fewer instructions, developers need to use more instructions. For example, did you know that [ARM usually doesn't have a division instruction](https://cseweb.ucsd.edu/classes/wi14/cse30-c/lectures/PI_WI_14_CSE30_lecture_8_post.pdf)? You have to divide by using [other instructions](https://stackoverflow.com/questions/19844575/how-to-do-division-in-arm). Since you're using other instructions to fake division, it ends up taking many cycles. This could even be slower than on a CISC instruction set.
```

Comment:
Replace did you know... with information straight-up for reasons mentioned above. Explain the division process briefly to avoid clicking link- simplify.

Reply:
I doubt it could be simplified in less than a paragraph in a way that everyone will understand. I also don't think including a whole paragraph explaining division is an efficient use of the reader's time. The article is about contrasting ARM with x86. Explaining efficient binary division would be a major shift in the subject.

I don't think it's really necessary either. If the reader wants to know why division is complicated, clicking the link is a trivial way of figuring that out. If they don't care to know, then the rest of the article won't suffer because of it.